packages/turf-boolean-crosses/index.js

```diff
@@ -0,0 +1,153 @@
+var helpers = require('@turf/helpers');
+var inside = require('@turf/inside');
+var lineIntersect = require('@turf/line-intersect');
+var polyToLinestring = require('@turf/polygon-to-linestring');
+var invariant = require('@turf/invariant');
+var getGeom = invariant.getGeom;
+var getGeomType = invariant.getGeomType;
+
+/**
+ * Boolean-Crosses returns True if the intersection results in a geometry whose dimension is one less than
+ * the maximum dimension of the two source geometries and the intersection set is interior to
+ * both source geometries.
+ *
+ * Boolean-Crosses returns t (TRUE) for only multipoint/polygon, multipoint/linestring, linestring/linestring, linestring/polygon, and linestring/multipolygon comparisons.
+ *
+ * @name booleanCrosses
+ * @param {Geometry|Feature<any>} feature1 GeoJSON Feature or Geometry
+ * @param {Geometry|Feature<any>} feature2 GeoJSON Feature or Geometry
+ * @returns {Boolean} true/false
+ * @example
+ * var cross = turf.booleanCrosses(feature1, feature2);
+ */
+
+module.exports = function (feature1, feature2) {
+    var type1 = getGeomType(feature1);
+    var type2 = getGeomType(feature2);
+    var geom1 = getGeom(feature1);
+    var geom2 = getGeom(feature2);
+
+    switch (type1) {
+    case 'MultiPoint':
+        switch (type2) {
+        case 'LineString':
+            return doMultiPointAndLineStringCross(geom1, geom2);
+        case 'Polygon':
+            return doesMultiPointCrossPoly(geom1, geom2);
+        }
+        throw new Error('feature2 ' + type2 + ' geometry not supported');
+    case 'LineString':
+        switch (type2) {
+        case 'LineString':
+            return doLineStringsCross(geom1, geom2);
+        case 'Polygon':
+            return doLineStringAndPolygonCross(geom1, geom2);
+        }
+        throw new Error('feature2 ' + type2 + ' geometry not supported');
+    default:
+        throw new Error('feature1 ' + type1 + ' geometry not supported');
```

Comment:
`type1` cannot be Point/Polygon/MultiLineString? or you are still working on this?